src/goose.rs

```diff
@@ -1163,31 +1208,32 @@ impl GooseClient {
     ///     /// A simple task that makes a GET request.
     ///     async fn get_function(client: &mut GooseClient) {
     ///         let response = client.get("/404").await;
-    ///         match &response {
+    ///         match &response.response {
     ///             Ok(r) => {
     ///                 // We expect a 404 here.
     ///                 if r.status() == 404 {
-    ///                     client.set_success(&GooseMethod::GET, "/404");
+    ///                     client.set_success(&response.request);
     ///                 }
     ///             },
     ///             Err(_) => (),
     ///         }
     ///     }
     /// ````
-    pub fn set_success(&mut self, method: &GooseMethod, path: &str) {
-        let request_name = self.get_request_name(path);
-        let mut update_request = GooseRawRequest::new(method.clone(), &request_name);
-        update_request.success = true;
-        // This is an updaate to a previously recorded statistic.
-        update_request.update = true;
-
-        self.send_to_parent(update_request);
+    pub fn set_success(&mut self, request: &GooseRawRequest) {
+        // Only send update if this was previously not a success.
+        if !request.success {
```

Comment:
Hmmm - would need to update the request as well then afterwards, because else .success() .fail() .success() would not do the right thing ...

Reply:
Excellent point! It also is doing an unnecessary copy because of this. Fixed to require a mutable pointer so we can modify in place, and just resend the modified request object to the parent.